File: TelerikUIExamples/TelerikUIExamples/StackedColumnChart.m
//
//  StackedColumnChart.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "StackedColumnChart.h"
#import <TelerikUI/TelerikUI.h>

@implementation StackedColumnChart
{
    TKChart *_chart;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        [self addOption:@"Stacked" selector:@selector(reloadData)];
        [self addOption:@"Stacked 100" selector:@selector(reloadData)];
        [self addOption:@"Clustered" selector:@selector(reloadData)];
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    _chart.allowAnimations = YES;
    [self.view addSubview:_chart];
    
    [self reloadData];
}

- (void)reloadData
{
    [_chart removeAllData];
    TKChartStackInfo *stackInfo = nil;
    if (self.selectedOption < 2) {
        TKChartStackMode stackMode = self.selectedOption == 0 ? TKChartStackModeStack : TKChartStackModeStack100;
        stackInfo = [[TKChartStackInfo alloc] initWithID:@(1) withStackMode:stackMode];
    }
    
    for (int i = 0; i < 4; i++) {
        NSMutableArray *points = [[NSMutableArray alloc] init];
        for (int j = 1; j < 8; j++) {
            [points addObject:[[TKChartDataPoint alloc] initWithX:@(j) Y:@(arc4random() % 100)]];
        }
        
        TKChartSeries *series = [[TKChartColumnSeries alloc] initWithItems:points];
        
        series.title = [NSString stringWithFormat:@"Series %d", i];
        series.stackInfo = stackInfo;
        series.selectionMode = TKChartSeriesSelectionModeSeries;
        [_chart addSeries:series];
    }
    [_chart reloadData];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInCSharp/Examples/Chart/StackedCharts/StackedColumnChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	public class StackedColumnChart: ExampleViewController
	{
		TKChart chart;

		public StackedColumnChart ()
		{
			this.AddOption ("Stacked", reloadData);
			this.AddOption ("Stacked 100", reloadData);
			this.AddOption ("No Stacking", reloadData);
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			this.reloadData (this, EventArgs.Empty);
		}

		void reloadData(object sender, EventArgs e) 
		{
			chart.RemoveAllData ();

			TKChartStackInfo stackInfo = null;
			if (this.SelectedOption < 2) {
				stackInfo = new TKChartStackInfo (new NSNumber(1), this.SelectedOption == 0 ? TKChartStackMode.Stack : TKChartStackMode.Stack100);
			}

			Random r = new Random();
			for (int i = 0; i < 4; i++) {
				List<TKChartDataPoint> list = new List<TKChartDataPoint>();
				for (int j = 0; j < 8; j++) {
					list.Add(new TKChartDataPoint(new NSNumber(j), new NSNumber(r.Next() % 100)));
				}

				TKChartColumnSeries series = new TKChartColumnSeries (list.ToArray ());
				series.Title = String.Format ("Series %d", i);
				series.StackInfo = stackInfo;
				series.SelectionMode = TKChartSeriesSelectionMode.Series;
				chart.AddSeries (series);
			}
		}
	}
}

